### ISSUE TITLE
(cbor) Introduce the writePOJO (or writeObject) with the element size

### ISSUE THREAD

**Issue by Kieun** [FasterXML/jackson-dataformats-binary#329]
CBOR has some variants to optimize the encoded output size to indicate the length of the elements.
Jackson CBOR generator has been supporting such feature by introducing some public methods.
But, there is no such variant method for **writeObject** or **writePOJO**.
In order to properly generate the optimized output (some implementation forces to optimize the output due to interoperability), it would be better to have such features in the CBOR generator.

So, new API signature seems to be

`public void writePOJO(Object pojo, int elementsToWrite) throws IOException`
or
`public void writePOJO(Object pojo, boolean makeDefiniteLength) throws IOException`

**Comment by cowtowncoder** [comment]
I don't think this is possible to implement at that level, unfortunately; `CBORGenerator` will essentially delegate back to `ObjectMapper` which does serialization. But this is not to say that it might not be possible to implement support at databind level. It would require significant changes to `BeanSerializer`, to determine if the number of properties is easy enough to reliably detect (may not be when filtering with `@JsonView` or alternatives is used, f.ex). But might be doable.

**Comment by cowtowncoder** [comment]
Overlapping with #3, fwtw.
